File: 202408/3016.py
class Solution:
    def minimumPushes(self, word: str) -> int:
        j = 0
        ans = 0
        for k, v in Counter(word).most_common():
            if j >= 24:
                ans += v * 4
            elif j >= 16:
                ans += v * 3
            elif j >= 8:
                ans += v * 2
            else:
                ans += v
            j += 1
        return ans

File: 202607/3016.cpp
class Solution {
public:
  int minimumPushes(string word) {
    vector<int> cnt(26);
    for (char c : word) {
      ++cnt[c - 'a'];
    }
    sort(begin(cnt), end(cnt));
    int ans = 0;
    int j = 0;
    int press = 0;
    for (int i = 25; i >= 0; --i) {
      if (cnt[i] == 0)
        break;
      if ((j++ % 8) == 0) {
        ++press;
      }
      ans += press * cnt[i];
    }
    return ans;
  }
};
